Sources/StreamChat/Controllers/MessageController/MessageController.swift

```diff
@@ -722,19 +722,78 @@ public class ChatMessageController: DataController, DelegateCallable, DataStoreP
     }
 
     /// Marks the thread read if this message is the root of a thread.
-    internal func markThreadRead(completion: ((Error?) -> Void)? = nil) {
+    public func markThreadRead(completion: ((Error?) -> Void)? = nil) {
         messageUpdater.markThreadRead(cid: cid, threadId: messageId) { error in
-            completion?(error)
+            self.callback {
+                completion?(error)
+            }
         }
     }
 
     /// Marks the thread unread if this message is the root of a thread.
-    internal func markThreadUnread(completion: ((Error?) -> Void)? = nil) {
+    public func markThreadUnread(completion: ((Error?) -> Void)? = nil) {
         messageUpdater.markThreadUnread(
             cid: cid,
             threadId: messageId
         ) { error in
-            completion?(error)
+            self.callback {
+                completion?(error)
+            }
+        }
+    }
+
+    /// Fetches the thread information of the message this controller manages.
+    /// Returns an error in case the message is not the root of a thread.
+    /// - Parameters:
+    ///   - replyLimit: The number of replies fetched.
+    ///   - participantLimit: The number of participants fetches.
+    ///   - completion: Returns the thread information if the message is the root of a thread.
+    public func loadThread(
+        replyLimit: Int? = nil,
+        participantLimit: Int? = nil,
+        completion: @escaping ((Result<ChatThread, Error>) -> Void)
+    ) {
+        var query = ThreadQuery(
+            messageId: messageId,
+            watch: false
```

Comment:
When do we watch threads if we set it false here?

Reply:
I see, we can discard this comment

Reply:
We watch threads in the Thread List. When opening the Thread, we are watching the replies. But in our `MessageController` there is no way to fetch thread information, so this additional action was added. But most likely it won't be used by most customers.